tests/serverpod_test_server/test_e2e/websocket_auth_test.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:async';
+
+import 'package:serverpod_test_client/serverpod_test_client.dart';
+import 'package:serverpod_test_server/test_util/config.dart';
+import 'package:serverpod_test_server/test_util/test_key_manager.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Given a websocket that requires authentication', () {
+    TestAuthKeyManager authKeyManager = TestAuthKeyManager();
+    late Client client;
+
+    setUp(() async {
+      client = Client(
+        serverUrl,
+        authenticationKeyManager: authKeyManager,
+      );
+    });
+
+    tearDown(() async {
+      client.close();
+    });
+
+    test('Access restricted endpoint without authentication', () async {
+      await client.openStreamingConnection(
+        disconnectOnLostInternetConnection: false,
+      );
+
+      // This should be ignored by the server as user isn't authenticated.
+      await client.signInRequired.sendStreamMessage(SimpleData(num: 666));
+      var streamingStream = client.signInRequired.stream.timeout(
+        const Duration(seconds: 2),
```

Comment:
Discussion(Knowledge sharing): We always try to stay away from making tests depend on timeouts. The reason for this is that these make tests brittle (dependent on execution time) and also might add an unnecessary delay to a test.

In this case, would it be enough to send another message to a different endpoint and validate that this response was received but the one that requires authentication was not?

Reply:
There are cases where we have to wait on a timeout. But in general we try to stay away from them.

Reply:
Same discussion for the other test.

Reply:
I agree with you, timeouts are un unwanted externality here. This was the most direct way of testing this that I found. If I merely call another endpoint, I'm not directly verifying that *this* endpoint is reacting as I expect.

But thinking about it I can send two messages, one before auth and one after. If the first element I receive back is the first I sent unauthenticated, then it is an error.